app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/tei_data/TEIDataPresenterImpl.java

```diff
@@ -225,43 +216,10 @@ public void onFollowUp(DashboardProgramModel dashboardProgramModel) {
 
     @Override
     public void onShareClick(View mView) {
-        PopupMenu menu = new PopupMenu(view.getContext(), mView);
-
-        menu.getMenu().add(Menu.NONE, Menu.NONE, 0, "QR");
-        if (mView.getResources().getBoolean(R.bool.sms_enabled)) {
-            menu.getMenu().add(Menu.NONE, Menu.NONE, 2, "SMS");
-        }
-        menu.getMenu().add(Menu.NONE, Menu.NONE, 2, "NFC");
-
-        menu.setOnMenuItemClickListener(item -> {
-            switch (item.getOrder()) {
-                case 0:
-                    view.analyticsHelper().setEvent(TYPE_SHARE, TYPE_QR, SHARE_TEI);
-                    Intent intent = new Intent(view.getContext(), QrActivity.class);
-                    intent.putExtra("TEI_UID", teiUid);
-                    view.showQR(intent);
-                    return true;
-                case 1:
-                    view.analyticsHelper().setEvent(TYPE_SHARE, TYPE_SMS, SHARE_TEI);
-                    Activity activity = view.getAbstractActivity();
-                    Intent i = new Intent(activity, SmsSubmitActivity.class);
-                    Bundle args = new Bundle();
-                    InputArguments.setEnrollmentData(args, dashboardModel.getCurrentEnrollment().uid());
-                    i.putExtras(args);
-                    activity.startActivity(i);
-                    return true;
-                case 2:
-                    view.analyticsHelper().setEvent(TYPE_SHARE, TYPE_NFC, SHARE_TEI);
-                    Intent intentNfc = new Intent(view.getContext(), NfcDataWriteActivity.class);
-                    intentNfc.putExtra("TEI_UID", teiUid);
-                    view.showQR(intentNfc);
-                    return true;
-                default:
-                    return true;
-            }
-        });
-
-        menu.show();
+        view.analyticsHelper().setEvent(TYPE_SHARE, TYPE_QR, SHARE_TEI);
+        Intent intent = new Intent(view.getContext(), QrActivity.class);
+        intent.putExtra("TEI_UID", teiUid);
+        view.showQR(intent);
```

Comment:
This should be on the view, as well as, using the AnalyticsHelper instead of the deprecated view.analyticsHelper()

Reply:
True. I'll fix it